src/preprocess/preprocess.go

```diff
@@ -258,40 +259,39 @@ func exposeAPIs_main(input string) string {
 		`\w+\(\)\.createElement\(\w+,\{onChange:this\.handleSaberStateChange\}\),`,
 		"")
 
-	utils.Replace(
+	// React Hook
+	utils.ReplaceOnce(
 		&input,
-		`;class \w+ extends (\w+)\(\).Component`,
-		`;Spicetify.React=${1}()${0}`)
+		`\w+=\(\w+,(\w+)\.lazy\)\(\(function\(\)\{return Promise\.resolve\(\)\.then\(\w+\.bind\(\w+,\w+\)\)\}\)\);`,
+		`${0}Spicetify.React=${1};`)
 
 	utils.Replace(
 		&input,
 		`"data-testid":`,
 		`"":`)
 
-	reAllAPIPromises := regexp.MustCompile(`await Promise.all\(\[([\w\(\)\.,]+?)\]\)([;,])`)
+	reAllAPIPromises := regexp.MustCompile(`return (\w+=\w+\.sent),\w+\.next=\d,(Promise.all\(\[(\w\.getSession\(\),)([\w\(\)\.,]+?)\]\))([;,])`)
 	allAPIPromises := reAllAPIPromises.FindAllStringSubmatch(input, -1)
 	for _, found := range allAPIPromises {
-		splitted := strings.Split(found[1], ",")
+		splitted := strings.Split(found[3] + found[4], ",")
 		if len(splitted) > 15 { // Actual number is about 24
 			re := regexp.MustCompile(`\w+\.(\w+)\(\)`)
-			code := "Spicetify.Platform = {"
+			// set t = e.sent, call Promise.all for APIs, then add Spicetify APIs to object
+			code := found[1] + ";" + found[2] + ".then(v => {Spicetify.Platform = {};"
 
+			i := 0
 			for _, apiFunc := range splitted {
 				name := re.ReplaceAllString(apiFunc, `${1}`)
 
 				if strings.HasPrefix(name, "get") {
 					name = strings.Replace(name, "get", "", 1)
 				}
-
-				code += name + ": await " + apiFunc + ","
+				code += "Spicetify.Platform[\"" + name + "\"] = v[" + fmt.Sprint(i) + "];"
+				i = i + 1
```

Comment:
In go, on a for range, you can just get the index by adding it to the for.
So instead of
```go
i := 0
for _, apiFunc := range splitted {
  name := re.ReplaceAllString(apiFunc, `${1}`)
  
  if strings.HasPrefix(name, "get") {
	  name = strings.Replace(name, "get", "", 1)
  }
  
  code += name + ": await " + apiFunc + ","
  code += "Spicetify.Platform[\"" + name + "\"] = v[" + fmt.Sprint(i) + "];"
  i = i + 1
}
```

```go
for apiFuncIndex, apiFunc := range splitted {
  name := re.ReplaceAllString(apiFunc, `${1}`)
  
  if strings.HasPrefix(name, "get") {
	  name = strings.Replace(name, "get", "", 1)
  }
  
  code += name + ": await " + apiFunc + ","
  code += "Spicetify.Platform[\"" + name + "\"] = v[" + fmt.Sprint(apiFuncIndex) + "];"

}
```

Reply:
Shows my inexperience with go.. Implementing!